src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -412,15 +414,16 @@ protected override LocalSymbol MakePatternVariable(DeclarationPatternSyntax node
                             scopeBinder: _scopeBinder,
                             nodeBinder: _enclosingBinder,
                             typeSyntax: node.Type,
-                            identifierToken: node.Identifier,
+                            identifierToken: designation.Identifier,
                             kind: LocalDeclarationKind.PatternVariable,
                             nodeToBind: nodeToBind,
                             forbiddenZone: null);
         }
         protected override LocalSymbol MakeOutVariable(DeclarationExpressionSyntax node, BaseArgumentListSyntax argumentListSyntax, SyntaxNode nodeToBind)
         {
             NamedTypeSymbol container = _scopeBinder.ContainingType;
-            var designation = (SingleVariableDesignationSyntax)node.Designation;
+            var designation = node.Designation as SingleVariableDesignationSyntax;
+            if (designation == null) return null;
```

Comment:
Please use braces even for single line ifs.  That is the coding guidelines we use.